tensorflow_datasets/structured/bbbp.py

```diff
@@ -0,0 +1,114 @@
+
+_CITATION = """\
+@article{doi:10.1021/ci300124c,
+author = {Martins, Ines Filipa and Teixeira, Ana L. and Pinheiro, Luis and Falcao, Andre O.},
+title = {A Bayesian Approach to in Silico Blood-Brain Barrier Penetration Modeling},
+journal = {Journal of Chemical Information and Modeling},
+volume = {52},
+number = {6},
+pages = {1686-1697},
+year = {2012},
+doi = {10.1021/ci300124c},
+    note ={PMID: 22612593},
+URL = { 
+        https://doi.org/10.1021/ci300124c
+    
+},
+eprint = { 
+        https://doi.org/10.1021/ci300124c
+    
+}
+}
+"""
+
+_DESCRIPTION = """\
+The Blood-brain barrier penetration (BBBP) dataset is extracted from a study on
+the modeling and prediction of the barrier permeability. As a membrane separating
+circulating blood and brain extracellular fluid, the blood-brain barrier blocks
+most drugs, hormones and neurotransmitters. Thus penetration of the barrier forms
+a long-standing issue in development of drugs targeting central nervous system.
+This dataset includes binary labels for over 2000 compounds on their permeability 
+properties. 
+
+The data file contains a csv table, in which columns below are used:
+     "name" - Name of the compound
+     "smiles" - SMILES representation of the molecular structure
+     "p_np" - Binary labels for penetration/non-penetration
+
+"""
+
+FEATURES = collections.OrderedDict([
+    ('name', tf.string),
+    ('p_np',  tfds.features.ClassLabel(num_classes = 2))
+])
+	
+	
+class BBBP(tfds.core.GeneratorBasedBuilder):
+    """BBBP dataset."""
+    NUM_CLASSES = 3
+    VERSION = tfds.core.Version('0.0.1')
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            # tfds.features.FeatureConnectors
+            features=tfds.features.FeaturesDict({
+	            'smiles': tf.string,
+	            'features': {name: dtype for name, dtype in FEATURES.items()}
+	        }),
+            supervised_keys=("features", "smiles"),
+            homepage="http://moleculenet.ai/datasets-1",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        bbbp_path = dl_manager.download_and_extract(BBBP_URL)
+        csv_path = os.path.join(bbbp_path, "BBBP.csv")
+        all_lines = tf.io.gfile.GFile(csv_path).read().split("\n")[1:-1]
+        all_lines = [','.join(line.split(',')[1:])[:-1].replace('"','') for line in all_lines]
+		
+        # Specify the splits
+        return [
+            tfds.core.SplitGenerator(
+                name=tfds.Split.TRAIN,
+                gen_kwargs={
+                    "records": all_lines,
+                }),
+        ]
+
+    def _generate_examples(self, records):
+        for i, row in enumerate(records):
+            elems = row.split(",")
+            features = {'name': ','.join(elems[:-2]), 'p_np': elems[-2]}
+            yield i, {
+                "features": features,
+                "smiles": elems[-1],
+            }
```

Comment:
Please add extra line after code ending and run `./oss_scripts/lint.sh tensorflow_datasets/image` and check if it gives 10/10 score